core/src/main/java/smashdudes/ecs/systems/HitDetectionSystem.java

```diff
@@ -83,6 +84,15 @@ private AttackResult hasEntityAttackedOther(Entity attacker, Entity attacked)
         AttackResult result = null;
         float largestCollisionArea = 0;
 
+        BodyBox blockBox = new BodyBox(otherBlock.blockBox);
+        blockBox.x = otherPos.position.x + blockBox.x - blockBox.width / 2;
+        blockBox.y = otherPos.position.y + blockBox.y - blockBox.height / 2;
+
+        if(otherBlock.isEnabled)
+        {
+            bodyBoxes.add(blockBox);
+        }
```

Comment:
This will cause the bodyBox list to infinitely grow, it never gets removed from the bodybox list and there will be duplicate elements. 

Find another way to test the shield-atackBox collision, I would suggest adding a unique check for it within the `AttackBox` loop.

Reply:
This does not infinitely add to a list, since the `bodyBoxes` array is generated within the method call.